projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/strategy/datajob/JobFieldStrategyByLastExecutionTime.java

```diff
@@ -0,0 +1,105 @@
+    * @param criteria Existing criteria containing already chained Predicates and sorting
+    * @param filter Provided filter containing information about needed sorting direction and filtering pattern.
+    * @throws GraphqlErrorException if the filter pattern contains invalid format or if the first unix time is after the second one
+    * @return The altered chained Predicates and comparator
+    */
+   @Override
+   public Criteria<V2DataJob> computeFilterCriteria(@NonNull Criteria<V2DataJob> criteria, @NonNull Filter filter) {
+      Predicate<V2DataJob> predicate = criteria.getPredicate();
+
+      if (filterProvided(filter)) {
+         predicate = predicate.and(dataJob -> {
+            var lastExecutionTime = getLastExecutionTime(dataJob);
+            var range = validateAndExtractDateRange(filter);
+
+            return lastExecutionTime != null &&
+                    !lastExecutionTime.isBefore(range.getFirst()) &&
+                    !lastExecutionTime.isAfter(range.getSecond());
+         });
+      }
+
+      return new Criteria<>(predicate, detectSortingComparator(filter, COMPARATOR_DEFAULT, criteria));
+   }
+
+   @Override
+   public Predicate<V2DataJob> computeSearchCriteria(@NonNull String searchStr) {
+      return dataJob -> false;
+   }
+
+   private Pair<OffsetDateTime, OffsetDateTime> validateAndExtractDateRange(Filter filter) {
+      String[] parsedStr = filter.getPattern().split(DATE_SEPARATOR);
+      if (parsedStr.length != 2) {
+         throw GraphqlErrorException.newErrorException()
+               .message(getStrategyName().getPath() + " does not contains valid date range.")
+               .build();
+      }
+      try {
+         var start = Integer.parseInt(parsedStr[0]);
+         var end = Integer.parseInt(parsedStr[1]);
+         var startDate = OffsetDateTime.ofInstant(Instant.ofEpochSecond(start), UTC_ZONE);
+         var endDate = OffsetDateTime.ofInstant(Instant.ofEpochSecond(end), UTC_ZONE);
+
+         if (startDate.isAfter(endDate)) {
+            throw GraphqlErrorException.newErrorException()
+                  .message("First date of " + getStrategyName().getPath() + " is after the second." +
+                        "The beginning date should be before the ending date")
+                  .build();
+         }
+         return Pair.of(startDate, endDate);
+      } catch (NumberFormatException | DateTimeException ex) {
+         throw GraphqlErrorException.newErrorException()
+               .message(getStrategyName().getPath() + " does not contains valid dates." +
+                     "Correct format is \"1619460302-1619460302\"")
+               .build();
+      }
+   }
+
+   private static OffsetDateTime getLastExecutionTime(V2DataJob dataJob) {
+      if (CollectionUtils.isEmpty(dataJob.getDeployments())) {
+         return null;
+      }
+      // TODO support for multiple deployments
+      return dataJob.getDeployments().stream().findFirst().orElseThrow().getLastExecutionTime();
```

Comment:
In already implemented code we perform a null check for deployments e.g:
https://github.com/vmware/versatile-data-kit/blob/f086df874af207e4f36447e50b7d7c0897e3af37/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/ExecutionDataFetcher.java#L81-L83
Perhaps it is worthwhile do do the same here?